token/src/state.rs

```diff
@@ -11,21 +11,26 @@ use solana_sdk::{
 };
 use std::mem::size_of;
 
-/// Represents a token type identified by its public key.  Accounts
-/// are associated with a specific token type and only accounts with
-/// matching types my inter-opt.
+/// Mint data.
 #[repr(C)]
 #[derive(Clone, Copy, Debug, Default, PartialEq)]
-pub struct Token {
+pub struct Mint {
     /// Optional owner, used to mint new tokens.  The owner may only
     /// be provided during mint creation.  If no owner is present then the mint
     /// has a fixed supply and no further tokens may be minted.
     pub owner: COption<Pubkey>,
     /// Number of base 10 digits to the right of the decimal place.
     pub decimals: u8,
+    /// Is `true` if this mint has been initialized
+    pub is_initialized: bool,
```

Comment:
Nit: I'd move this to the front of the struct

Reply:
And same for the other structs?

Reply:
Yeah, I had it there for aesthetic reasons originally but it packs tighter to have the `u8`'s where they are now

Reply:
Originally I didn't have these and was going to just check some known value in the structure but that's pretty weak and for `Mint` there isn't anything definitive that I can check.  The 'Multisig' structure increased by one byte but I poked the `is_initialized` member into structure padding for `Mint` and `Account`.  

On a separate note, we could use packed structs to save a few bytes, I can run a few tests to see how much of a hit that is performance-wise

Reply:
> I had it there for aesthetic reasons originally but it packs tighter to have the `u8`'s where they are now

Ah, ok that explanation is good enough for me :)